packages/x-date-pickers/src/AdapterMomentJalaali/AdapterMomentJalaali.ts

```diff
@@ -173,15 +173,11 @@ export class AdapterMomentJalaali
   };
 
   public isSameYear = (value: Moment, comparing: Moment) => {
-    // `isSame` seems to mutate the date on `moment-jalaali`
-    // @ts-ignore
-    return value.clone().isSame(comparing, 'jYear');
+    return value.jYear() === comparing.jYear();
   };
 
   public isSameMonth = (value: Moment, comparing: Moment) => {
-    // `isSame` seems to mutate the date on `moment-jalaali`
-    // @ts-ignore
-    return value.clone().isSame(comparing, 'jMonth');
+    return value.jYear() === comparing.jYear() && value.jMonth() === comparing.jMonth();
```

Comment:
This is the main problem that was causing the behavior, because it is called here:

https://github.com/mui/mui-x/blob/6c4308ff3f9d23f6f36d84b74fb89d3c2ed29176/packages/x-date-pickers/src/DateCalendar/useCalendarState.tsx#L71-L74

And it caused the focus to be changed to the first day of the current month.